mautic/mautic
# Fix PHP Notice - Undefined index: ct

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
This PR fixed a lot of warnings in Mautic.

`[2021-09-10 07:13:28] mautic.NOTICE: PHP Notice - Undefined index: ct - in file /path/app/bundles/PageBundle/Controller/PublicController.php - at line 450 {"redirectId":"25cac6cf84a40ed70cf5a4741","logger":"[object] (Symfony\\Bridge\\Monolog\\Logger: {})","redirectModel":"[object] (Mautic\\PageBundle\\Model\\RedirectModel: {})","redirect":"[object] (Mautic\\PageBundle\\Entity\\Redirect: Mautic\\PageBundle\\Entity\\Redirect with ID #2212)","url":"https://url/","query":[]} []`

Email trackable links are usually in format:


`https://urlofmautic.com/r/110a69363fd96ed4e28d8e662?ct=YTo1OntzOjY6InNvdXJjZSI7YToyOntpOjA7coxNDoiY2FtcGFpZ24uZXZlbnQiO2k6MTtpOjI1MTg7fXM6NToiZW1haWwiO2k6MTc5O3M6NDoic3RhdCI7czoyMjojBjMDY5MjI1OTk1NjMzMTkMDc1OCI7czo0OiJsZWFkIjtzOjY6IjI2MzI1NCI7czo3OiJjaGFubmVsIjthOjE6e3M6NToiZW1haWwiO2k6MTc5O319`

But bots (https://github.com/mautic/mautic/issues/9798) transform it to 

`https://urlofmautic.com/r/110a69363fd96ed4e28d8e662?somethingelse=cccccc`

Then looks like for redirect links required always ct param to be track clicks to contact. 




#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create email with link
3. Send email
4. Copy link from email
5. Remove ct param from url and add another one (for exmaple `https://urlofmautic.com/r/110a69363fd96ed4e28d8e662?somethingelse=cccccc`)
6. Before PR it return warning in logs. Now should works properly




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10418"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
